generator_v3.js

```diff
@@ -140,90 +147,93 @@ var generator_v3 = function(layout, difficulty, bingoList)
+			{
+				if (reactants.some(r=> goalCandidate.catalyst.includes(r)))
 				{
-					// If it is get a new goal
-					console.log("reactant/catalyst between: " + goalCandidate.name + " and " + currentSheet[indexes[z]].name);
-					cont = false;
-					break;
+					// If it is, get a new goal
+					console.log("reactants for: " + goalCandidate.name);
+					continue GoalGen;
 				}
-
-				// Compare all the tags of the current goal with the tags of the current one on the sheet
-				if (goalCandidate.tags != null && currentSheet[indexes[z]].tags != null)
+			}
+			// Check if the goal generated is a reactant for anything already on the sheet
+			if (typeof goalCandidate.reactant !== 'undefined')
+			{
+				// If it does, check to see if it's already on the sheet
+				if (catalysts.some(r=> goalCandidate.reactant.includes(r)))
 				{
-					for (var x = 0, lenX = goalCandidate.tags.length; x < lenX; x++)
-					{
-						for (var y = 0, lenY = currentSheet[indexes[z]].tags.length; y < lenY; y++)
-						{
-							// If they're the same tag
-							if (goalCandidate.tags[x].name == currentSheet[indexes[z]].tags[y].name)
-							{
-								// If the current goal has a tag that disallows placement on the same line and they are on the same line
-								if (goalCandidate.tags[x].line == false && isOnSameLine(indexes[i], indexes[z]))
-								{
-									console.log("cannot be on same line: " + goalCandidate.name + " and " + currentSheet[indexes[z]].name);
-									cont = false;
-								}
-								// Otherwise just increase the tag count
-								else
-								{
-									tagCount[x]++;
-								}
-							}
-						}
-					}
+					// If it is, get a new goal
+					console.log("catalyst for: " + goalCandidate.name);
+					continue GoalGen;
 				}
 			}
 
-			// If the tag count is higher than the max that tag should have, don't continue
-			if (cont && goalCandidate.tags != null)
+			// If the goal candidate contains tags that cannot be on the same line as other goals with that tag
+			if (goalCandidate.tags.some(r=> r.line == false))
 			{
-				for (var x = 0, len = goalCandidate.tags.length; x < len; x++)
+				// Go through every goal currently on the sheet
+				for (var z=0; z < i; z++)
 				{
-					if (tagCount[x] >= goalCandidate.tags[x].max[difficulty - 1])
+					// If both goals have tags AND are on the same line
+					if (goalCandidate.tags != null && currentSheet[indexes[z]].tags != null && isOnSameLine(indexes[i], indexes[z]))
 					{
-						console.log(goalCandidate.tags[x].name + " max reached with " + tagCount[x] + " on the board");
-						cont = false;
-						break;
+						// If both goals have the same tag
+						if (currentSheet[indexes[z]].tags.some(r=> r.line == false && goalCandidate.tags.some(s=> r.name === s.name)))
+						{
+							console.log("Cannot be on same line: " + goalCandidate.name + " and " + currentSheet[indexes[z]].name);
+							continue GoalGen;
+						}
 					}
 				}
-			}
+			}			
```

Comment:
Unnecessary whitespace added at the end of this line.

Reply:
You do love your whitespace